### PR TITLE
Add Akismet API key to thank you page

### PR DESCRIPTION
**Pull request by jboland88**
## Proposed Changes

* This diff adds the Akismet API key to the thank you page following an Akismet purchase

## Testing Instructions

* Navigate to `http://calypso.localhost:3000/checkout/akismet/thank-you/ak_plus_yearly_1` while logged in
* You should see your API key on the thank you page with a button to copy it

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx [hunk 1]
```diff
@@ -72,13 +77,18 @@ const AkismetCheckoutThankYou: FunctionComponent< AkismetCheckoutThankYouProps >
...
 
+				<p className="akismet-checkout-thank-you__api-key-header">
+					{ __( 'Your Akismet API Key is:', 'akismet-thank-you' ) }
+				</p>
+				<ThankYouAPIKeyClipboard />
+
 				<Button
 					primary
 					busy={ isLoading }
 					href="https://akismet.com/account/"
```
Comment (CodeyGuyDylan) [inline client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx:86]: Was the plan still to link to the Calypso purchase here? 0-as-1204210417785446/1204285311458080/f

File: client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx [hunk 2]
```diff
@@ -87,4 +97,48 @@ const AkismetCheckoutThankYou: FunctionComponent< AkismetCheckoutThankYouProps >
 	);
 };
 
+function ThankYouAPIKeyClipboard() {
+	const translate = useTranslate();
+	const [ isCopied, setCopied ] = useState( false );
```
Comment (CodeyGuyDylan) [inline client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx]: I think by convention, the setting function should be named `setIsCopied`

File: client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx [hunk 3]
```diff
@@ -87,4 +97,48 @@ const AkismetCheckoutThankYou: FunctionComponent< AkismetCheckoutThankYouProps >
...
+	if ( isError ) {
+		return null;
+	}
+
+	const akismetApiKey = data ?? '';
+	const keyInputSize = akismetApiKey ? akismetApiKey.length + 5 : 0;
+
+	return (
+		<div
+			className={ 'akismet-checkout-thank-you__key-clipboard' + ( isLoading ? ' loading' : '' ) }
```
Comment (CodeyGuyDylan) [inline client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx]: For multiple classnames I like importing
```tsx
import classNames from 'classnames';
```
So they can be added in a comma separated list. Not required by any stretch though

File: client/my-sites/checkout/checkout-thank-you/style.scss [hunk 4]
```diff
@@ -881,6 +881,46 @@ main.akismet-checkout-thank-you {
...
+			color: var(--color-text);
+			border: none;
+		}
+
+		.button {
+			min-width: 100px;
+			font-size: 1.25rem;
+		}
+
+		@include breakpoint-deprecated( ">480px" ) {
```
Comment (CodeyGuyDylan) [inline client/my-sites/checkout/checkout-thank-you/style.scss]: I see that the deprecated breakpoint is being used in this file quite a bit, but it looks like this value is the same as if you did this
```scss
@include break-mobile;
```

**Review by CodeyGuyDylan — CHANGES_REQUESTED** [review]
I left three minor non-blockers, the main question was whether or not we will be linking to Calypso purchase instead of `/account`

**Comment by and1can** [comment]
I am testing in localhost and I do not see an API key. I am not saying the issue is caused by the pull request. I am wondering if there is any configuration that needs to happen on my end. I am an OSS Citizen. I do see a logged error that says `Unauthorized Error`. 

checkout_akismet
unauthorized_error

**Comment by CodeyGuyDylan** [comment]
> I am testing in localhost and I do not see an API key. I am not saying the issue is caused by the pull request. I am wondering if there is any configuration that needs to happen on my end. I am an OSS Citizen. I do see a logged error that says Unauthorized Error.

Were you logged in during checkout? If not, I can see being logged out and going straight to the thank you page causing this. If you are logged out during checkout, it creates a temporary user for you once you complete the checkout, and in that case I would think you would avoid getting that error 🤔 

If that is the case, we could probably handle an authentication error better by just not displaying the whole API key section

If you were logged in, I am not entirely sure why that would happen. Maybe Josh has more insight

**Comment by CodeyGuyDylan** [comment]
I can confirm I got the same error when not logged in @jboland88. When I went through the userless checkout instead of going straight to the page, the API key loaded correctly. So I think the main thing to do here is just not render the entire API key section if the key does not load.

I also noticed the placeholder looks a bit off when the API key is loading, should be some space between it and the CTA
image

File: client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx (hunk 1 above)
Comment (jboland88) [reply client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx:86]: I think that would be ideal, however, I was running into the issue of not knowing the correct purchase ID in this context. I wasn't finding much luck in obtaining the purchase ID without having to modify the route and how/ when it gets called after the purchase :) TLDR; I think it's too complex for the current iteration.

File: client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx (hunk 3 above)
Comment (jboland88) [reply client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx]: TIL about `classnames`!

File: client/my-sites/checkout/checkout-thank-you/style.scss (hunk 4 above)
Comment (jboland88) [reply client/my-sites/checkout/checkout-thank-you/style.scss]: Good catch - thanks!

File: client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx [hunk 5]
```diff
@@ -87,4 +95,56 @@ const AkismetCheckoutThankYou: FunctionComponent< AkismetCheckoutThankYouProps >
...
+
+	return (
+		<>
+			<p className="akismet-checkout-thank-you__api-key-header">
+				{ __( 'Your Akismet API Key is:', 'akismet-thank-you' ) }
+			</p>
+			<div
+				className={ classNames(
+					'akismet-checkout-thank-you__key-clipboard',
+					isLoading ? ' loading' : ''
```
Comment (CodeyGuyDylan) [inline client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx]: I don't think it makes a difference but you shouldn't need the space here anymore 😄 
```suggestion
					isLoading ? 'loading' : ''
```

**Comment by jboland88** [comment]
Thanks for testing and noting an issue @and1can 👏 

@CodeyGuyDylan 
I think we can assume the thank you page should only show when logged in since the CTA is directed to the account page. I've updated the route. Additionally, I've updated the output so that the whole API key section is not shown if there is some other kind of loading error.

File: client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx (hunk 5 above)
Comment (jboland88) [reply client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx]: yep yep - applied!

**Review by CodeyGuyDylan — APPROVED** [review]
Tests great now! Thanks for the changes 😄

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7905158

Hi @jboland88, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Your Akismet API Key is:`

Thank you in advance!

File: client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx [hunk 6]
```diff
@@ -87,4 +95,56 @@ const AkismetCheckoutThankYou: FunctionComponent< AkismetCheckoutThankYouProps >
 	);
 };
 
+function ThankYouAPIKeyClipboard() {
+	const translate = useTranslate();
+	const [ isCopied, setIsCopied ] = useState( false );
+	const { data, isError, isLoading } = useAkismetKeyQuery();
+
+	useEffect( () => {
```
Comment (and1can) [inline client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx:103]: note to self: `setTimeout` serves to allow notification to display for ~4 seconds and then remove notification

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7905158).

**Pull request merged**